components/txn_types/src/types.rs

```diff
@@ -253,6 +253,37 @@ pub enum MutationType {
     Other,
 }
 
+/// A row mutation.
+#[derive(Debug, Clone)]
+pub enum AtomicMutation {
+    /// Put `Value` into `Key` with ttl, overwriting any existing value.
+    Put((Key, Value), Option<u64>),
+    /// Delete `Key`.
+    Delete(Key),
+    /// Set a lock on `Key`.
+    Lock(Key),
+    /// Put `Value` into `Key` if `Key` does not yet exist.
+    ///
+    /// Returns `kvrpcpb::KeyError::AlreadyExists` if the key already exists.
+    Insert((Key, Value)),
+    /// Check `key` must be not exist.
+    ///
+    /// Returns `kvrpcpb::KeyError::AlreadyExists` if the key already exists.
+    CheckNotExists(Key),
```

Comment:
These are not used in `AtomicStore`

Reply:
it's used in https://github.com/tikv/tikv/blob/master/src/storage/txn/commands/atomic_store.rs#L36 and https://github.com/tikv/tikv/blob/master/src/storage/txn/commands/atomic_store.rs#L40

Reply:
You can safely remove those.

Reply:
Because they are never constructed.